enhancements/machine-api/aws-imds-v2-support.md

```diff
@@ -0,0 +1,276 @@
+
+## Proposal
+
+- Add new options into install-config for the OCP installation program which will be passed as metadata service settings during virtual machines creation
+- Extend [AWSMachineProviderConfig](https://github.com/openshift/api/blob/1a6fa2913810101176a1d776f899fc4781b3fa50/machine/v1beta1/types_awsprovider.go#L12) by adding metadata service parameters
+- Pass metadata service parameters to AWS during new machines creation
+
+### User Stories
+
+- As an OpenShift cluster administrator, I would like to enforce IMDSv2 usage during an OCP cluster deployment procedure.
+- As an OpenShift cluster administrator, I would like to enforce IMDSv2 usage for newly created machines in my cluster via MachineSets configuration.
+- As an OpenShift cluster administrator, I would like to change IMDS interaction settings for newly created machines in my cluster.
+
+### API Extensions
+
+#### AWSMachineProviderConfig changes
+
+MachineAPI design mostly replicates parameters which [AWS API](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_InstanceMetadataOptionsRequest.html) consumes.
+
+```go
+type AWSMachineProviderConfig struct {
+    // Existing fields will not be modified
+    ...
+    // MetadataServiceOptions allows users to configure instance metadata service interaction options.
+    // If nothing specified, default AWS IMDS settings will be applied.
+    // https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_InstanceMetadataOptionsRequest.html
+    // +optional
+    MetadataServiceOptions *MetadataServiceOptions `json:"metadataServiceOptions,omitempty"`
+}
+```
+
+```go
+
+type InstanceMetadataHTTPTokens string
+
+const (
+    // InstanceMetadataHTTPTokensRequired enforces sending of a signed token header with any instance metadata retrieval (GET) requests.
+    // Enforces IMDSv2 usage.
+    InstanceMetadataHTTPTokensRequired = "required"
+    // InstanceMetadataHTTPTokensOptional allows IMDSv1 usage along with IMDSv2
+    InstanceMetadataHTTPTokensOptional = "optional"
+)
+
+// MetadataServiceOptions defines the options available to a user when configuring
+// Instance Metadata Service (IMDS) Options.
+type MetadataServiceOptions struct {
+    // HTTPTokens determines whether or not the host requires the use of authentication when interacting with the metadata service.
+    // When using authentication, this enforces v2 interaction method (IMDSv2) with the metadata service.
+    // Default: optional
+    // +kubebuilder:validation:Enum=required;optional
+    // +optional
+    HTTPTokens InstanceMetadataHTTPTokens `json:"httpTokens,omitempty"`
+    // HTTPPutResponseHopLimit is the desired HTTP PUT response hop limit for instance metadata requests.
+	// By default, the response to PUT requests has a response hop limit (time to live, TTL) of 1 at the IP protocol level.
+	// For example, you might need a larger hop limit for backward compatibility with container services running on the instance.
```

Comment:
I thought we had concluded that all containers accessing the metadata need to be on host networking? Therefore the hop will be 1 always?